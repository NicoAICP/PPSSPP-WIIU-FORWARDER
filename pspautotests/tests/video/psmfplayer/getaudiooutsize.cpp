#include "shared.h"
#include <pspthreadman.h>

void testGetAudioOutSize(const char *title, SceUID *psmf) {
	int result = scePsmfPlayerGetAudioOutSize(psmf);
	if (result >= 0) {
		checkpoint("%s: OK: %d", title, result);
	} else {
		checkpoint("%s: Failed: %08x", title, result);
	}
}

extern "C" int main(int argc, char *argv[]) {
	loadPsmfPlayer();

	sceMpegInit();
	checkpointNext("Init");

	checkpointNext("Players:");
	testGetAudioOutSize("  Initial", createPsmfPlayerInitial());
	testGetAudioOutSize("  Standby", createPsmfPlayerStandby());
	testGetAudioOutSize("  Deleted", createPsmfPlayerDeleted());
	// Crashes.
	//testGetAudioOutSize("  NULL", NULL);
	SceUID *psmfPlaying = createPsmfPlayerPlaying();
	testGetAudioOutSize("  Playing", psmfPlaying);
	testGetAudioOutSize("  Twice", psmfPlaying);
	psmfPlaying = createPsmfPlayerPlaying();
	SceUID psmf2 = *psmfPlaying;
	testGetAudioOutSize("  Copy", &psmf2);
	testGetAudioOutSize("  Finished", createPsmfPlayerFinished());

	sceMpegFinish();
	unloadPsmfPlayer();
	return 0;
}