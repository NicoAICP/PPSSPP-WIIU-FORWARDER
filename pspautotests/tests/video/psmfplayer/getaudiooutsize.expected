[x] Init

[x] Players:
[r]   Initial: OK: 8192
[r]   Standby: OK: 8192
[r]   Deleted: Failed: 80616001
[r]   Playing: OK: 8192
[x]   Twice: OK: 8192
[r]   Copy: OK: 8192
[r]   Finished: OK: 8192
